### PR TITLE
[ci/mac] Add nightly Mac arm64 (M1/M2) wheel builds to CI

### PR DESCRIPTION
**Pull request by krfricke**
## Why are these changes needed?

Mac Silicon (arm64, M1, M2) wheels currently have to be built manually in our release process. After AWS introduced M1 machines in EC2, we can now add a buildkite job that builds the artifacts on every commit, like other mac builds.

This PR introduces a new buildkite job for the Mac arm64 wheel builds. It also makes minor modifications to the existing build scripts to work with macOS Big Sur and arm64.

In the future, we should migrate the existing mac builds to use Big Sur as well, so we can get rid of some of the multiplexing logic we currently have in the build scripts.

This PR will also build artifacts for release branches, so we can get rid of the manual M1 wheel build step in the release process.

## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
    - [ ] I've added any new APIs to the API Reference. For example, if I added a 
           method in Tune, I've added it in `doc/source/tune/api/` under the 
           corresponding `.rst` file.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Comment by zhe-thoughts** [comment]
Will wait for a committer approval first (@wuisawesome or @pcmoritz )

**Comment by cadedaniel** [comment]
This is an awesome improvement that has large impact on the release manager workload. Nice work @krfricke!!

**Review by cadedaniel — APPROVED** [review]

File: python/build-wheel-macos-arm64.sh
Comment (wuisawesome) [inline python/build-wheel-macos-arm64.sh:25]: I feels like if we move these 2 lines out of the if statement, we can avoid a lot of the new branches being added to this script?

**Comment by krfricke** [comment]
Note that the build job is succeeding, but S3 access to the artifact bucket seems to be gone for both linux arm64 and mac arm64 instances

File: ci/env/install-dependencies.sh
Comment (wuisawesome) [inline ci/env/install-dependencies.sh:98]: Can we do this block within the `case` statements above? really it seems like we should just have one case statement that sets all these variables for each platform.

**Review by wuisawesome — COMMENTED** [review]
Left a couple of comments about cleaning up some of the branches in the script/avoiding adding new branches, but generally lgtm

**Comment by krfricke** [comment]
Wheels are built and uploaded:

Screenshot 2023-03-30 at 2 07 28 PM

File: python/build-wheel-macos-arm64.sh
Comment (krfricke) [reply python/build-wheel-macos-arm64.sh:25]: Let's clean this up in a follow-up. We should generally look if we can improve these scripts.

For now I want to avoid changing too much of the existing CI scripts before the branch cut

**Review by zhe-thoughts — APPROVED** [review]
Approved for merging to master

**Review by wuisawesome — APPROVED** [review]

**Pull request merged**
